core/build.gradle.kts

```diff
@@ -0,0 +1,26 @@
+plugins {
+    id(BuildPlugins.kotlin)
+    id(BuildPlugins.javaLibrary)
+    id(ScriptPlugins.pureKotlin)
+}
+
+tasks.register<Jar>(Core.configArtifactName) {
+    dependsOn("testClasses")
+    archiveBaseName.set("${project.name}-test")
+    from(sourceSets["test"].output.classesDirs)
+}
+
+val testConfig = configurations.create(Core.configName) {
+    extendsFrom(configurations["testCompile"])
+}
+
+artifacts {
+    add(Core.configName, tasks.named<Jar>(Core.configArtifactName))
```

Comment:
There's no clear indication if this will impact performance in anyway, but, it's a nice workaround rather than having two modules (core and core-test), which share the same responsibilities to the project overall anyway. In my opinion it doesn't seem like something that should be worried about too much considering we're running tests in a debugging environment.